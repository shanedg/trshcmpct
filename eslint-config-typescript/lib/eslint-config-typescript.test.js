const eslint = require('eslint');

const ESLint = eslint.ESLint;

describe('eslint-config-typescript', () => {
  let eslintApi;

  beforeAll(() => {
    eslintApi = new ESLint({
      overrideConfigFile: 'lib/eslint-config-typescript.js',
      useEslintrc: false,
    });
  });

  describe('calculated config snapshots', () => {
    let config;

    beforeAll(async () => {
      // This file doesn't actually have to exist and the path doesn't matter
      // since we override the config file and useEslintrc is false.
      config = await eslintApi.calculateConfigForFile('anyFile.ts');
    });

    it('parserOptions', () => {
      expect(config.parserOptions).toMatchSnapshot();
    });

    it('rules', () => {
      expect(config.rules).toMatchSnapshot();
    });

    it('settings', () => {
      expect(config.settings).toMatchSnapshot();
    });
  });

});
