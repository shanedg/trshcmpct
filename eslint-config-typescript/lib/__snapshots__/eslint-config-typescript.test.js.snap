// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-typescript calculated config snapshots parserOptions 1`] = `
{
  "sourceType": "module",
}
`;

exports[`eslint-config-typescript calculated config snapshots rules 1`] = `
{
  "@typescript-eslint/adjacent-overload-signatures": [
    "error",
  ],
  "@typescript-eslint/await-thenable": [
    "error",
  ],
  "@typescript-eslint/ban-ts-comment": [
    "error",
  ],
  "@typescript-eslint/ban-types": [
    "error",
  ],
  "@typescript-eslint/no-array-constructor": [
    "error",
  ],
  "@typescript-eslint/no-empty-function": [
    "error",
  ],
  "@typescript-eslint/no-empty-interface": [
    "error",
  ],
  "@typescript-eslint/no-explicit-any": [
    "warn",
  ],
  "@typescript-eslint/no-extra-non-null-assertion": [
    "error",
  ],
  "@typescript-eslint/no-extra-semi": [
    "error",
  ],
  "@typescript-eslint/no-floating-promises": [
    "error",
  ],
  "@typescript-eslint/no-for-in-array": [
    "error",
  ],
  "@typescript-eslint/no-implied-eval": [
    "error",
  ],
  "@typescript-eslint/no-inferrable-types": [
    "error",
  ],
  "@typescript-eslint/no-loss-of-precision": [
    "error",
  ],
  "@typescript-eslint/no-misused-new": [
    "error",
  ],
  "@typescript-eslint/no-misused-promises": [
    "error",
  ],
  "@typescript-eslint/no-namespace": [
    "error",
  ],
  "@typescript-eslint/no-non-null-asserted-optional-chain": [
    "error",
  ],
  "@typescript-eslint/no-non-null-assertion": [
    "warn",
  ],
  "@typescript-eslint/no-this-alias": [
    "error",
  ],
  "@typescript-eslint/no-unnecessary-type-assertion": [
    "error",
  ],
  "@typescript-eslint/no-unnecessary-type-constraint": [
    "error",
  ],
  "@typescript-eslint/no-unsafe-argument": [
    "error",
  ],
  "@typescript-eslint/no-unsafe-assignment": [
    "error",
  ],
  "@typescript-eslint/no-unsafe-call": [
    "error",
  ],
  "@typescript-eslint/no-unsafe-member-access": [
    "error",
  ],
  "@typescript-eslint/no-unsafe-return": [
    "error",
  ],
  "@typescript-eslint/no-unused-vars": [
    "warn",
    {
      "argsIgnorePattern": "^_",
    },
  ],
  "@typescript-eslint/no-var-requires": [
    "error",
  ],
  "@typescript-eslint/prefer-as-const": [
    "error",
  ],
  "@typescript-eslint/prefer-namespace-keyword": [
    "error",
  ],
  "@typescript-eslint/require-await": [
    "error",
  ],
  "@typescript-eslint/restrict-plus-operands": [
    "error",
  ],
  "@typescript-eslint/restrict-template-expressions": [
    "error",
  ],
  "@typescript-eslint/triple-slash-reference": [
    "error",
  ],
  "@typescript-eslint/unbound-method": [
    "error",
  ],
  "constructor-super": [
    "off",
  ],
  "getter-return": [
    "off",
  ],
  "import/default": [
    "off",
  ],
  "import/named": [
    "off",
  ],
  "import/namespace": [
    "off",
  ],
  "import/no-named-as-default-member": [
    "off",
  ],
  "no-array-constructor": [
    "off",
  ],
  "no-const-assign": [
    "off",
  ],
  "no-dupe-args": [
    "off",
  ],
  "no-dupe-class-members": [
    "off",
  ],
  "no-dupe-keys": [
    "off",
  ],
  "no-empty-function": [
    "off",
  ],
  "no-extra-semi": [
    "off",
  ],
  "no-func-assign": [
    "off",
  ],
  "no-implied-eval": [
    "off",
  ],
  "no-import-assign": [
    "off",
  ],
  "no-loss-of-precision": [
    "off",
  ],
  "no-new-symbol": [
    "off",
  ],
  "no-obj-calls": [
    "off",
  ],
  "no-redeclare": [
    "off",
  ],
  "no-setter-return": [
    "off",
  ],
  "no-this-before-super": [
    "off",
  ],
  "no-undef": [
    "off",
  ],
  "no-unreachable": [
    "off",
  ],
  "no-unsafe-negation": [
    "off",
  ],
  "no-unused-vars": [
    "off",
  ],
  "no-var": [
    "error",
  ],
  "prefer-const": [
    "error",
  ],
  "prefer-rest-params": [
    "error",
  ],
  "prefer-spread": [
    "error",
  ],
  "require-await": [
    "off",
  ],
  "valid-typeof": [
    "off",
  ],
}
`;

exports[`eslint-config-typescript calculated config snapshots settings 1`] = `
{
  "import/extensions": [
    ".ts",
    ".cts",
    ".mts",
    ".tsx",
    ".js",
    ".jsx",
  ],
  "import/external-module-folders": [
    "node_modules",
    "node_modules/@types",
  ],
  "import/parsers": {
    "@typescript-eslint/parser": [
      ".ts",
      ".cts",
      ".mts",
      ".tsx",
    ],
  },
  "import/resolver": {
    "node": {
      "extensions": [
        ".ts",
        ".cts",
        ".mts",
        ".tsx",
        ".js",
        ".jsx",
      ],
    },
    "typescript": {
      "project": "tsconfig.json",
    },
  },
}
`;
